api/client/src/main/java/org/projectnessie/client/api/NessieApiV2.java

```diff
@@ -28,4 +28,10 @@ public interface NessieApiV2 extends NessieApiV1 {
   GetRepositoryConfigBuilder getRepositoryConfig();
 
   UpdateRepositoryConfigBuilder updateRepositoryConfig();
+
+  /** Delete a branch or a tag. */
+  DeleteReferenceBuilder deleteReference();
```

Comment:
The old `deleteBranch` and `assignBranch` methods are still available, right? Wouldn't this be confusing to end users? Wasn't it possible to override the existing methods with covariant return types?

Reply:
Not sure about covariant return types. I guess we still have to maintain method signatures to avoid breaking older clients compiled with java 8... Also, old `getAndDelete()` methods have more specific return types.

Reply:
That's said but I guess we don't have the choice then. 👍

Reply:
WDYT about overriding the functions in `NessieApiV2` and mark them as deprecated there?

Reply:
Changing functions would break existing code (ABI)

Reply:
+1, very good suggestion.

Reply:
Deprecated V1 java client assign/delete methods in the latest push.